R/mergecategoriescontainingranges.R

```diff
@@ -176,7 +177,22 @@ MergeRangeCategories <- function(input.data,
                           index = 2)
     label.data$first.value <- first.value
     label.data$second.value <- second.value
-    label.data <- label.data[order(label.data$first.value), ]
+
+    # Disambiguate first.value when sorting to ensure that 
+    # open-ended labels are sorted correctly. Otherwise causes
+    # a problem when same number appears twice in labels and
+    # when the categories are initially out of order. For example
+    # if the first two labels are "18 to 24" and "Under 18".
+    sort.values = label.data$first.value
+    for (j in 1L:nrow(label.data)) {
```

Comment:
`seq_len` preferred instead of `1L:val` in loops, the latter can execute code in loop when there are no rows, while the former protects against it.